### ISSUE TITLE
 Issue with my Angular Ionic app where the SCSS files aren't being properly tracked, which is causing the Heat Maps UI to be destroyed. 

### ISSUE THREAD

**Issue by tuhinkolese** [microsoft/clarity#545]
Hello everyone, 

I am currently facing an issue while implementing ms-clarity in an Angular website and ionic Native (Android application) using Capacitor. While the clarity dashboard is successfully capturing data, the SCSS/CSS is not loading properly.

I kindly request your assistance in proposing a solution to this problem. Any input or advice would be greatly appreciated. 

Angular Ionic App Scss not getting Tracked properly and Heat Maps UI getting Destroyed."

**Comment by ibradwan** [comment]
Hi tuhinkolese,

Could you please elaborate more whether you're using the ClarityJs script within your website or you're using the Clarity SDK for Ionic apps?

**Comment by tuhinkolese** [comment]
Hi Ibrahim,
In our project, we utilize the Clarity SDK for Android. For the web portion
of the project, we use the ClarityJS Script which is placed in the
index.html head tag.
For further reference, I have shared a code snippet for the web portion
below.


<head> ...........
...........

<script type="text/javascript">
(function(c,l,a,r,i,t,y){
c[a]=c[a]||function(){(c[a].q=c[a].q||[]).push(arguments)};
t=l.createElement(r);t.async=1;t.src="https://www.clarity.ms/tag/"+i;
y=l.getElementsByTagName(r)[0];y.parentNode.insertBefore(t,y);
})(window, document, "clarity", "script", "project_id");
</script>

</head>

Project Framework: We are using Ionic with Angular along with Capacitor for
the Native part of the project.

Tuhin Kole
Associate Tech Lead
StockEdge |  <http://stockedge.com/>stockedge.com
<http://web.stockedge.com/>
+919732202664
Sector V, Kolkata





<https://twitter.com/PurviJain97>

<https://www.linkedin.com/in/purvi-jain-484987167/>





<https://twitter.com/PurviJain97>

<https://www.linkedin.com/in/purvi-jain-484987167/>

**Comment by ibradwan** [comment]
Well, you should need to do so if you're using the [Ionic package](https://www.npmjs.com/package/cordova-clarity), because the package will capture both native and web content in the same session.

**Comment by tuhinkolese** [comment]
Well, I removed the script from HTML and used the ionic package but in the web, ClarityPlugin is not defined.
![image](https://github.com/microsoft/clarity/assets/131441998/82b7ae35-c1cb-4fdb-b6ba-84235ab38268)

**Comment by tuhinkolese** [comment]
![image](https://github.com/microsoft/clarity/assets/131441998/6ae76d20-dfb1-44b4-9acb-6ebbf07a9e8a)

**Comment by ibradwan** [comment]
Yes that's a known issue with Capacitor + Ionic cordova based plugins. Please apply this condition and see:
![image](https://github.com/microsoft/clarity/assets/67283130/71682a25-3158-4d9d-839d-ce517802ffe9)

**Comment by tuhinkolese** [comment]
For Android implementations

![image](https://github.com/microsoft/clarity/assets/131441998/c22423fd-cd1b-4e5a-982e-1a38aced079a)

**Comment by przemekRBC** [comment]
@ibradwan We also face this issue with Clarity Ionic SDK package regarding SCSS styles. We use Angular as a fronted framework. Recordings seem to work correctly, however styles don't work.

Is there any temporary workaround for this or should we wait for Ionic Clarity SDK package update?

![Image](https://github.com/user-attachments/assets/93bf8bac-a5a9-473d-93fd-646c202ada59)

**Comment by przemekRBC** [comment]
@kiranraj1430 Sure, this is the code which we invoke on application startup:

![Image](https://github.com/user-attachments/assets/b51fb290-025f-4322-b14b-2ab33a702986)

**Comment by kiranraj1430** [comment]
> [@kiranraj1430](https://github.com/kiranraj1430) Sure, this is the code which we invoke on application startup:
> 
> ![Image](https://github.com/user-attachments/assets/b51fb290-025f-4322-b14b-2ab33a702986)

Looks good, but in my case i faced 2 issues

1: One is scss not rendered properly and
2:  another when i return back from any component the page are conflicting like UI stop rendering and page got shrink or squeezed. Due to that i didn't added ms clarity for production. My preference is wait for next update and add it in your app. 

Check the 2nd issue is causing in your app properly, i got the issue when i testing multiple times on page routing.

My tech: Angular + Tailwind UI, capacitor android app

**Comment by przemekRBC** [comment]
@kiranraj1430 Clarity initialisation code is being executed in ngOnInit method of app.component.ts.
As you described, routing doesn't work for me too. In our app scenario, whole Clarity video only shows dashboard page (which is first to  show up after app launch) and we can see that user is interacting with different pages based on some unstyled dialogs poping up (which are coming from other pages than the main dashboard page):

![Image](https://github.com/user-attachments/assets/bee3b73b-acab-4a09-92f4-7b5b57ae3a67)

**Comment by kiranraj1430** [comment]
> [@kiranraj1430](https://github.com/kiranraj1430) Clarity initialisation code is being executed in ngOnInit method of app.component.ts. As you described, routing doesn't work for me too. In our app scenario, whole Clarity video only shows dashboard page (which is first to show up after app launch) and we can see that user is interacting with different pages based on some unstyled dialogs poping up (which are coming from other pages than the main dashboard page)

Check properly, sometime 💯 there will be a routing issue or component render issue will be there. Sometimes it won't show for us but it have issue. Best solution is wait for next update of this plugin. Since i am waiting for more than 3 months. 


Atleast they should mention which file or which place we need to paste their configuration but they given a simple doc code to paste in a app.

**Comment by kiranraj1430** [comment]
![Image](https://github.com/user-attachments/assets/2547b05a-5005-4124-b92b-24d1517c0d2d)
Stupid plugin of the core, i need to find a different code to run this plugin lol,  when it gives success message but in the clarity dashboard I do not able to see live visitors. Please provide if you have proper docs else don't publish the plugin. we are not testing rats to test your buggy code.

**Comment by kiranraj1430** [comment]
> [@kiranraj1430](https://github.com/kiranraj1430) Clarity initialisation code is being executed in ngOnInit method of app.component.ts. As you described, routing doesn't work for me too. In our app scenario, whole Clarity video only shows dashboard page (which is first to show up after app launch) and we can see that user is interacting with different pages based on some unstyled dialogs poping up (which are coming from other pages than the main dashboard page)

Hey could you share your tech used?
Angular capacitor without ionic ? or angular + ionic + capacitor ?

**Comment by przemekRBC** [comment]
@kiranraj1430 We use Angular 18 with Ionic 8 and Capacitor 6.

**Comment by brunoinds** [comment]
Hi guys! I had a insight about the styles traking issue... As Ionic uses <ion-page> as a wrapper for the entire page, it uses their own scroll event handler, and it works such as a virtual scroll, so, when a external plugin tries to track the page, it does not work, because they can't get the scroll event, so that is why the styles are broken...

**Comment by tuhinkolese** [comment]
Awesome, thanks for the update!
